bundles/framework/announcements/tool/AnnouncementsTool.js

```diff
@@ -0,0 +1,367 @@
+                        break;
+                    }
+                }
+            }
+            return toolPlugin;
+        },
+
+        fetchAnnouncements: function () {
+            jQuery.ajax({
+                type: 'GET',
+                dataType: 'json',
+                url: Oskari.urls.getRoute('Announcements'),
+                success: (pResp) => {
+                    this.announcements = pResp.data;
+                },
+                error: function (jqXHR, textStatus) {
+                }
+            });
+        },
+
+        _getAnnouncementsSelection: function () {
+            var me = this,
+                announcementsSelection = {};
+            var pluginValues = me.getPlugin().getSelectedAnnouncements();
+            if (pluginValues.announcements) {
+                announcementsSelection.announcements = pluginValues.announcements;
+            }
+            return announcementsSelection;
+        },
+
+        /**
+        * Get values.
+        * @method getValues
+        * @public
+        *
+        * @returns {Object} tool value object
+        */
+        getValues: function () {
+            var me = this;
+
+            if (me.state.enabled) {
+                var pluginConfig = { id: this.getTool().id, config: this.getPlugin().getConfig() };
+                var announcementsSelection = me._getAnnouncementsSelection();
+
+                if (announcementsSelection && !jQuery.isEmptyObject(announcementsSelection)) {
+                    pluginConfig.config.announcements = announcementsSelection.announcements;
+                }
+                return {
+                    configuration: {
+                        mapfull: {
+                            conf: {
+                                plugins: [pluginConfig]
+                            }
+                        }
+                    }
+                };
+            } else {
+                return null;
+            }
+        },
+
+        /**
+         * "Sends" an event, that is, notifies other components of something.
+         *
+         * @method _sendEvent
+         * @param {String} eventName the name of the event
+         * @param {Whatever} eventData the data we want to send with the event
+         */
+        _sendEvent: function (eventName, eventData) {
+            var eventBuilder = Oskari.eventBuilder(eventName),
+                evt;
+
+            if (eventBuilder) {
+                evt = eventBuilder(eventData);
+                this.__sandbox.notifyAll(evt);
+            }
+        },
+
+        /**
+        * Sends an event to notify interested parties that the announcements have changed.
+        *
+        * @method _sendAnnouncementsChangedEvent
+        * @param {Object} announcements the changed announcement
+        */
+        _sendAnnouncementsChangedEvent: function (announcements) {
+            this._sendEvent('Announcements.AnnouncementsChangedEvent', announcements);
```

Comment:
I can see this event useful as part of the announcements functionality. I don't see it as a good practice if it's only used to transfer information between this publisher tool and the plugin. You already have and use a reference to the plugin on this tool so you could just call a function on that plugin instead.